Mathlib/Analysis/SpecialFunctions/Log/ENNRealLog.lean

```diff
@@ -0,0 +1,228 @@
+/-
+Copyright (c) 2024 Damien Thomine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone
+-/
+import Mathlib.Analysis.SpecialFunctions.Pow.NNReal
+import Mathlib.Topology.Instances.EReal
+
+/-!
+# Extended nonnegative real logarithm
+We define `log` as an extension of the logarithm of a positive real
+to the extended nonnegative reals `ℝ≥0∞`. The function takes values
+in the extended reals `EReal`, with `log 0 = ⊥` and `log ⊤ = ⊤`.
+
+## Main definitions
+- `log`: The extension of the real logarithm to `ℝ≥0∞`.
+- `log_OrderIso`, `log_equiv`: `log` seen respectively
+as an order isomorphism and an homeomorphism.
+
+## Main Results
+- `log_strictMono`: `log` is increasing;
+- `log_injective`, `log_surjective`, `log_bijective`: `log` is
+  injective, surjective, and bijective;
+- `log_mul_add`, `log_pow`, `log_pow`: `log` satisfy
+the identities `log (x * y) = log x + log y` and `log (x * y) = y * log x`
+(with either `y ∈ ℕ` or `y ∈ ℝ`).
+
+## Tags
+ENNReal, EReal, logarithm
+-/
+namespace ENNReal
+
+/-! ### Definition -/
+section Definition
+
+/--
+The logarithm function defined on the extended nonnegative reals `ℝ≥0∞`
+to the extended reals `EReal`. Coincides with the usual logarithm function
+and with `Real.log` on positive reals, and takes values `log 0 = ⊥` and `log ⊤ = ⊤`.
+Conventions about multiplication in `ℝ≥0∞` and addition in `EReal` make the identity
+`log (x * y) = log x + log y` unconditional. --/
+noncomputable def log (x : ℝ≥0∞) : EReal :=
+  if x = 0 then ⊥
+    else if x = ⊤ then ⊤
+    else Real.log (ENNReal.toReal x)
+
+@[simp]
+theorem log_zero : log 0 = ⊥ := by simp [log]
+
+@[simp]
+theorem log_one : log 1 = 0 := by simp [log]
+
+@[simp]
+theorem log_top : log ⊤ = ⊤ := by simp [log]
+
+theorem log_pos_real {x : ℝ≥0∞} (h : x ≠ 0) (h' : x ≠ ⊤) :
+    log x = Real.log (ENNReal.toReal x) := by simp [log, h, h']
+
+theorem log_pos_real' {x : ℝ≥0∞} (h : 0 < x.toReal) :
+    log x = Real.log (ENNReal.toReal x) := by
+  simp [log, Ne.symm (ne_of_lt (ENNReal.toReal_pos_iff.1 h).1),
+    ne_of_lt (ENNReal.toReal_pos_iff.1 h).2]
+
+theorem log_of_nNReal {x : NNReal} (h : x ≠ 0) :
+    log (ENNReal.ofReal x) = Real.log x := by simp [log, h]
```

Comment:
This statement is weird, because you are going through the map to reals and then to ennreals. It would be more natural to formulate it through the coercion from nnreal to ennreal (and then you could mark it as simp).

Reply:
Done. Thanks.